sdg/outputs/OutputGeoJson.py

```diff
@@ -0,0 +1,263 @@
+import os
+import json
+import copy
+from urllib.request import urlopen
+import sdg
+import pandas as pd
+from sdg.outputs import OutputBase
+
+class OutputGeoJson(OutputBase):
+    """Output SDG data/metadata in GeoJson disaggregated by region."""
+
+
+    def __init__(self, inputs, schema, output_folder='_site', translations=None,
+        geojson_file='regions.geojson', name_property='name', id_property='id',
+        id_column='GeoCode', output_subfolder='regions', filename_prefix='indicator_',
+        exclude_columns=None, id_replacements=None):
+        """Constructor for OutputGeoJson.
+
+        Parameters
+        ----------
+
+        Inherits all the parameters from OutputBase, plus the following:
+
+        geojson_file : string
+            A path to a GeoJSON file (remote or local) which contains all of the
+            "geometries" for the regions to include. Each region should have an
+            id and a name, in properties (see name_property and id_property).
+        name_property : string
+            The property in the geometry file which contains the region's name.
+        id_property : string
+            The property in the geometry file which contains the region's id.
+        id_column : string
+            The name of a column in the indicator data which corresponds to the
+            id that is in the "id_property" of the geometry file. This serves to
+            "join" the indicator data with the geometry file.
+        output_subfolder : string
+            A folder beneath 'geojson' to put the files. The full path will be:
+            [output_folder]/geojson/[output_subfolder]/[indicator_id].geojson
+        filename_prefix : string
+            A prefix added before the indicator id to construct a filename for
+            each geojson file.
+        exclude_columns : list
+            A list of strings, each a column name in the indicator data that
+            should not be included in the disaggregation. This is typically
+            for any columns that mirror the region referenced by the id column.
+        id_replacements : dict
+            An optional for with replacements to apply to the values in
+            the id_column. This is typically used if another column exists which
+            "mirrors" what would be in an id column, to avoid duplicate work.
+            For example, maybe a "Region" column exists with the names of the
+            regions as values. This can be used to "map" those region names to
+            geocodes, and save you the work of maintaining a separate id column.
+        """
+        if translations is None:
+            translations = []
+        if exclude_columns is None:
+            exclude_columns = []
+        if id_replacements is None:
+            id_replacements = {}
```

Comment:
Possible to initialise these in the constructor parameters?

Reply:
I would have thought so, and tried that at first. Apparently that's a Python gotcha. Method parameters should never be given defaults that are mutable types, like lists and dicts. If you do that, Python treats them like static "class properties", and every instance of the class shares the same value. Took me a lot of debugging to figure that out.

Reply:
Makes sense